### ISSUE TITLE
Allow customization of naming patterns for resources created with install.instance

### ISSUE THREAD

**Issue by sobjornstad** [tfsaggregator/aggregator-cli#126]
# Feature Request

**Is your feature request related to a problem? Please describe.**
Our organization has standard naming conventions for resources in Azure. Being able to provide only a single resource name to prefix all the resources the Aggregator creates (functions app, storage account, App Insights, App Service plan) is not sufficient to meet these naming conventions, which require specific prefixes for each type of resource.

**Describe the solution you would like**
Add options to individually name each of the resources created by the installation, or at least to alter them after the fact and have further operations continue to work. While we're at it, removing the requirement to have `aggregator` suffixed to the name might be nice (this one isn't an issue for me, but it might bother others).

**Describe alternatives you have considered**
I cannot think of any other way to solve this, but perhaps there would be some hacks possible if I knew more about how the Aggregator worked. Please let me know if you know of them.

**Comment by giuliov** [comment]
Thanks Soren for sharing your requirements. I fully understand the scenario and it would be easy to implement (naming algorithm is in `InstanceName.cs` and `instance-template.json`.
What blocks me from coding is that removing the `aggregator` string is a breaking change. Surfacing breaking changes is hard: even advertising using a big bold font is not sufficient.
Any suggestion for managing the transition is welcome.

**Comment by BobSilent** [comment]
I would provide a default behaviour, with the old prefix and keep the command line arguments for the default users.
And describe handling of the template file for experienced users, they do not provide the command line arguments but e.g. a "from naming template" argument, with the possibility to provide each name separately

**Comment by giuliov** [comment]
I like the idea of a _naming template_

I found four places where names are set
```cs
// src\aggregator-cli\Instances\InstanceName.cs:9-10
    const string resourceGroupPrefix = "aggregator-";
    const string functionAppSuffix = "aggregator";

// src\aggregator-cli\Instances\instance-template.json:63
    "storageAccountName": "[concat('aggregator', uniquestring(resourceGroup().id))]",

// src\aggregator-cli\Instances\AggregatorInstances.cs:189
    string deploymentName = SdkContext.RandomResourceName("aggregator", 24);

// src\aggregator-cli\Mappings\AggregatorMappings.cs:41
    && s.ConsumerInputs.GetValue("url","").IndexOf("aggregator.azurewebsites.net") > 8);
```
The first three set resource name while creating, so easy peasy.
The latter is used to find all WebHook Subscriptions in Azure DevOps.

Looks doable

**Comment by sobjornstad** [comment]
I like the naming-template idea as well, and was definitely thinking a new command-line option, not a change to the existing one. I don't see any reason to change the default behavior to remove *aggregator* -- it's probably more helpful than anything for people who aren't in a complex enterprise environment.

**Comment by giuliov** [comment]
Feature implemented in new [0.9.11](https://github.com/tfsaggregator/aggregator-cli/releases/tag/v0.9.11) release. @sobjornstad, would you like to have a look?

**Comment by sobjornstad** [comment]
Will do! I just got back from vacation -- I'll try to have a look at it this week.

**Comment by sobjornstad** [comment]
This is looking good! I'm having an issue mapping rules right now, but I'm pretty sure that's just our Azure configuration. Otherwise, it redeployed as desired with a naming template.

I'm puzzled about one thing, though: even though `--resourceGroup` has to be explicitly specified when using a naming template, one is still required to put a non-empty `ResourceGroupPrefix` or `ResourceGroupSuffix` into the naming template, and then one must specify only a part of the actual resource group name for `--resourceGroup` (so that the correct name gets put together by the prefix/suffix and the part supplied on the command line). What is the purpose of having prefix/suffix configuration options for resource groups at all when the resource group name won't be automatically generated anyway?

Not hard to get around, but it doesn't make much sense to me.

**Comment by giuliov** [comment]
It is due to some assumption in some code. I will explore if I can get rid of those.

**Comment by giuliov** [comment]
Removed the limitation in [0.9.13](https://github.com/tfsaggregator/aggregator-cli/releases/tag/v0.9.13). Now you can leave `ResourceGroupPrefix` and `ResourceGroupSuffix` out.

**Comment by giuliov** [comment]
Released in [0.9.14](https://github.com/tfsaggregator/aggregator-cli/releases/tag/v0.9.14)
